packages/google_sign_in/google_sign_in_web/lib/src/button_configuration.dart

```diff
@@ -0,0 +1,202 @@
+
+  /// The minimum button width, in pixels.
+  ///
+  /// The maximum width is 400 pixels.
+  final double? minimumWidth;
+
+  /// The pre-set locale of the button text.
+  ///
+  /// If not set, the browser's default locale or the Google session user's
+  /// preference is used.
+  ///
+  /// Different users might see different versions of localized buttons, possibly
+  /// with different sizes.
+  final String? locale;
+}
+
+/// The type of button to be rendered.
+///
+/// https://developers.google.com/identity/gsi/web/reference/js-reference#type
+enum GSIButtonType {
+  /// A button with text or personalized information.
+  standard,
+
+  /// An icon button without text.
+  icon;
+}
+
+const Map<GSIButtonType, id.ButtonType> _idType =
+    <GSIButtonType, id.ButtonType>{
+  GSIButtonType.icon: id.ButtonType.icon,
+  GSIButtonType.standard: id.ButtonType.standard,
+};
+
+/// The theme of the button to be rendered.
+///
+/// https://developers.google.com/identity/gsi/web/reference/js-reference#theme
+enum GSIButtonTheme {
+  /// A standard button theme.
+  outline,
+
+  /// A blue-filled button theme.
+  filledBlue,
+
+  /// A black-filled button theme.
+  filledBlack;
+}
+
+const Map<GSIButtonTheme, id.ButtonTheme> _idTheme =
+    <GSIButtonTheme, id.ButtonTheme>{
+  GSIButtonTheme.outline: id.ButtonTheme.outline,
+  GSIButtonTheme.filledBlue: id.ButtonTheme.filled_blue,
+  GSIButtonTheme.filledBlack: id.ButtonTheme.filled_black,
+};
+
+/// The size of the button to be rendered.
+///
+/// https://developers.google.com/identity/gsi/web/reference/js-reference#size
+enum GSIButtonSize {
+  /// A large button (about 40px tall).
+  large,
+
+  /// A medium-sized button (about 32px tall).
+  medium,
+
+  /// A small button (about 20px tall).
+  small;
+}
+
+const Map<GSIButtonSize, id.ButtonSize> _idSize =
+    <GSIButtonSize, id.ButtonSize>{
+  GSIButtonSize.large: id.ButtonSize.large,
+  GSIButtonSize.medium: id.ButtonSize.medium,
+  GSIButtonSize.small: id.ButtonSize.small,
+};
+
+/// The button text.
+///
+/// https://developers.google.com/identity/gsi/web/reference/js-reference#text
+enum GSIButtonText {
+  /// The button text is "Sign in with Google".
+  signinWith,
+
+  /// The button text is "Sign up with Google".
+  signupWith,
+
+  /// The button text is "Continue with Google".
+  continueWith,
+
+  /// The button text is "Sign in".
+  signin;
+}
+
+const Map<GSIButtonText, id.ButtonText> _idText =
+    <GSIButtonText, id.ButtonText>{
+  GSIButtonText.signinWith: id.ButtonText.signin_with,
+  GSIButtonText.signupWith: id.ButtonText.signup_with,
+  GSIButtonText.continueWith: id.ButtonText.continue_with,
+  GSIButtonText.signin: id.ButtonText.signin,
+};
+
+/// The button shape.
+///
+/// https://developers.google.com/identity/gsi/web/reference/js-reference#shape
+enum GSIButtonShape {
+  /// The rectangular-shaped button.
+  rectangular,
+
+  /// The circle-shaped button.
+  pill;
+  // Does this need circle and square?
```

Comment:
Maybe make this a github issue?

Reply:
I'll create an internal issue, this is a Google designed API.

Reply:
Opened b/276957876